src/dimwits/math.hpp

```diff
@@ -0,0 +1,113 @@
+namespace math {
+
+template< typename T >
+constexpr double e = 2.71828182845904523536028747135266249775724709369995;
+
+struct Exp{
```

Comment:
I wish these had some sort of comment stating what they did. 

Yes, I could read the source code. Yes the comments may become out of date. But at least a one-liner describing what is done can save myself and others an hour or more trying to figure out what you have created. 

It doesn't have to be extensive, just something giving someone an idea of what is being done. How do I know you aren't making up this algorithm? (I don't believe you are.) Do I need to become familiar with ways of implementing an exponential (or log) function to know what is going on? Sure, it would be helpful if I knew all that stuff, but it shouldn't be a pre-requisite to using this or to approving it.